### DISCUSSION TITLE
tailwind.config.jsのカスタマイズについて

### DISCUSSION THREAD

**Discussion by kagomen** [first-contributions-ja/first-contributions-ja.github.io#59] [90: 知識・知見の共有]
## 概要
tailwind.config.jsのカスタマイズについて、ここで話し合いや情報共有できればと思って立てました🎨

## 関連するイシュー
[各自でTailwind CSSを書いていくので、Webページが大体完成したらCSSの整理を行う #62](https://github.com/first-contributions-ja/first-contributions-ja.github.io/issues/62)

**Comment by kagomen** [comment]
使用するカラーを完全に縛るかどうか
[Tailwind CSS のカスタマイズ方法 -Zenn](https://zenn.dev/yohei_watanabe/books/c0b573713734b9/viewer/04642b)

**Reply by pss-aileen** [reply]
なるほど！
`theme > color` で定義すると定義した色だけになるんですね！
TailwindのCSSを使いつつ、カスタムカラーを追加したい場合は `theme > extend > colors` なんですね！

**Reply by kagomen** [reply]
まとめてくださり、ありがとうございます🙏
そうみたいです！
今回のプロジェクトだと赤・白・黒だけで構成できそうな感じもしますし、`theme > color`で縛っても良い気もしましたー！

**Comment by kazzyfrog** [comment]
なんとなく、以下のように進めていくと、Tailwindの理解深められそうな気がしています！

1. Tailwindを使って、デザインカンプ通りにUIを実装
2. `tailwind.config.js` をカスタマイズして、より柔軟に管理しやすくする
3. CSS変数や、必須級のTailwind系ライブラリを使う
4. shadcn/uiを導入する

**Reply by kagomen** [reply]
ロードマップ……！
ありがとうございます🙏✨

**Comment by kagomen** [comment]
kazzyfrogさんの上記のロードマップに則って、こちらのスレッドの対象を「Tailwind全般」から「tailwind.config.jsのカスタマイズ」に変更しても良いかなと思いました！

> CSS変数や、必須級のTailwind系ライブラリを使う
> shadcn/uiを導入する

ここらへんは情報量が膨大になる気がするので、configのカスタマイズが落ち着いてきたら適宜新しいスレッドを立てるという感じで……！

**Reply by kazzyfrog** [reply]
良いと思います！

**Comment by kagomen** [comment]
設定すること、あまりない気がする！
→今のところカラーの設定くらい？

あとは（必要なら）ブレークポイントを変えるくらいでしょうか？

**Reply by pss-aileen** [reply]
思ったより設定すること、今少なそうですね...笑
ブレイクポイントも変えなくて良いような気がしています...！🥹
[Responsive Design - Tailwind CSS](https://tailwindcss.com/docs/responsive-design)
こちらにしたがう感じで良いのかなと
